test/FileSystem.spec.js

```diff
@@ -0,0 +1,79 @@
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+const sinon = require('sinon');
+
+const FileSystem = require('../bin/FileSystem');
+
+const spy = sinon.spy;
+
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+describe('Filesystem class', () => {
+
+  describe('writeFile static method', () => {
+
+    let driverMock;
+
+    beforeEach(() => {
+      driverMock = {
```

Comment:
an alternative that wouldn't work for mocking out the driver is to have an in-memory implementation, as the tests need to trigger error conditions that cannot be driven from the outside like the `writeFileAsync` promise being rejected